gwu-libraries/sfm-ui
# Fixes #1009; adds configurable cookie consent popup

To test:

- Set `sfm-docker` to branch https://github.com/gwu-libraries/sfm-docker/tree/t1009-cookie-consent-envvars (in PR https://github.com/gwu-libraries/sfm-docker/pull/26 )
- Change `SFM_ENABLE_COOKIE_CONSENT` to `True` in `.env` file.
- Customize `SFM_COOKIE_CONSENT_HTML` and `SFM_COOKIE_CONSENT_BUTTON_TEXT`.  Suggested values for GW:
```
SFM_ENABLE_COOKIE_CONSENT=True
SFM_COOKIE_CONSENT_HTML=GW uses cookies and other technology on our websites to improve your website experience and to better understand how you use our websites. Visit GW's <a href="https://www.gwu.edu/privacy-notice" target="_blank" aria-label="Website privacy notice, opens in new window">Website Privacy Notice</a> to learn more about how GW uses cookies and, if you should choose, how to disable them. I understand that by clicking "I consent" and continuing to use this website, I agree to GW's use of cookies.
SFM_COOKIE_CONSENT_BUTTON_TEXT=I consent
```
- View the application and verify that the popup appears.  Click "I Consent".
- Verify that upon viewing the application again in the same browser, the popup does not appear.
- Change `SFM_ENABLE_COOKIE_CONSENT` to `False` in `.env` file.
- Open application in an incognito window - or clear cookies.
- Stop and restart the UI docker container
- Verify that the cookie consent popup does not appear.
- Build the documentation and verify the changes to the `Installation and Configuration` page
